MovingBlocks/DestinationSol
# GCI - Implementing JSON Schema Validation

# Description
GCI task by Torpedo
This PR replaces the old JSON library with org.json and implements schema validation using everit-org.
This PR is continued in #364 as this contains auto formatting of all project files while #364 only contains schema validation changes. 
# Schema Files to be added
**All schema files are to be added to engine\src\main\resources\assets\schemas**
- schemaColorsConfig,json -> core\assets\configs\colorsConfig
- schemaPlayerSpawnConfig ->playerSpawnConfig.json files
- schemaAbilitiesConfig -> core\assets\configs\abilitiesConfig
- schemaCollisionMesh -> engine:asteroids, engine:miscCollisionMeshes
- schemaSolNames -> engine:planetNamesConfig, engine:systemNamesConfig
- schemaStartingStation -> startingStation.json
- schemaTypes -> core\assets\items\types
- schemaShield -> core\assets\items\shields
- schemaGun -> core\assets\items\guns
- schemaAbilityCharges -> core\assets\items\abilityCharges
- schemaClip -> core\assets\items\Clips
- schemaEngine -> engine files
- schemaMazeConfig -> core\assets\configs\mazeConfig,json
- schemaPlanetsConfig -> core\assets\configs\planetsConfig.json
- schemaSystemConfig -> core\assets\configs\systemConfig.json
- schemaSpecialEffectConfig -> core\assets\configs\specialEffectConfig.json
- schemaProjectileConfig -> core\assets\configs\projectileConfig.json

# Schema Files added
- schemaHullConfig.json - #357 
- schemaArmorConfig.json - #357 

# Testing
- Load more schema files
- Start the game
- Play around
- Wait for crash
- If the game crashes, check console for error, if not, play more and have fun!

# Outstanding Work
- Including and testing more schema files.
